packages/preferences/src/browser/preference-tree-model.ts

```diff
@@ -227,6 +227,17 @@ export class PreferenceTreeModel extends TreeModelImpl {
         }
     }
 
+    getNodeFromPreferenceId(id: string): Preference.LeafNode | undefined {
+        if (this.root) {
+            for (const node of new TopDownTreeIterator(this.root)) {
+                if (Preference.LeafNode.is(node) && node.preferenceId === id) {
+                    return node;
+                }
+            }
+        }
+        return undefined;
+    }
+
```

Comment:
I believe that this can be made simpler if you make the `getGroupName` method of `PreferenceTreeGenerator` public and then do this:

```suggestion
    getNodeFromPreferenceId(preferenceId: string): Preference.LeafNode | undefined {
         const expectedGroup = this.treeGenerator.getGroupName(preferenceId.split('.'));
         const expectedId = `${expectedGroup}@${preferenceId}` // could combine this line and the previous as a utility on the PreferenceTreeGenerator: 'getNodeId(preferencId: string): string'
         return this.getNode(expectedId);
    }
```

The only reason ID's can't be computed completely programmatically is that an arbitrary set of preferences are in two groups: `Commonly Used` and whatever group they belong to 'naturally', but this procedure should always generate the correct ID pointing to the 'natural' location of a preference.